tensorflow/python/keras/applications/efficientnet_weight_update_util.py

```diff
@@ -0,0 +1,307 @@
+    use_ema: Bool, use the ExponentialMovingAverage resuolt in ckpt or not 
+    model_name_tf: str, the name of model in ckpt.
+
+  Returns:
+    String for the name of weight as in ckpt file.
+
+  Raises:
+    ValueError if keras_block does not show up in keras_name
+  """
+
+  if f'{keras_block}' not in keras_name:
+    raise ValueError(f'block name {keras_block} not found in {keras_name}')
+
+  # all blocks in the first group will not have expand conv and bn
+  is_first_blocks = (keras_block[5]=='1')
+
+  tf_name = [model_name_tf, tf_block]
+
+  # depthwide conv
+  if 'dwconv' in keras_name:
+    tf_name.append('depthwise_conv2d')
+    tf_name.append('depthwise_kernel')
+
+  # conv layers
+  if is_first_blocks:
+    # first blocks only have one conv2d
+    if 'project_conv' in keras_name:
+      tf_name.append('conv2d')
+      tf_name.append('kernel')
+  else:
+    if 'project_conv' in keras_name:
+      tf_name.append('conv2d_1')
+      tf_name.append('kernel')
+    elif 'expand_conv' in keras_name:
+      tf_name.append('conv2d')
+      tf_name.append('kernel')
+      
+  # squeeze expansion layers 
+  if '_se_' in keras_name:
+    if 'reduce' in keras_name:
+      tf_name.append('se/conv2d')
+    elif 'expand' in keras_name:
+      tf_name.append('se/conv2d_1')
+
+    if 'kernel' in keras_name:
+      tf_name.append('kernel')
+    elif 'bias' in keras_name:
+      tf_name.append('bias')
+
+  # batch normalization layers 
+  if 'bn' in keras_name:
+    if is_first_blocks:
+      if 'project' in keras_name:
+        tf_name.append('tpu_batch_normalization_1')
+      else:
+        tf_name.append('tpu_batch_normalization')
+    else:
+      if 'project' in keras_name:
+        tf_name.append('tpu_batch_normalization_2')
+      elif 'expand' in keras_name:
+        tf_name.append('tpu_batch_normalization')
+      else:
+        tf_name.append('tpu_batch_normalization_1')
+
+    for x in ['moving_mean', 'moving_variance', 'beta', 'gamma']:
+      if x in keras_name:
+        tf_name.append(x)
+  if use_ema:
+    tf_name.append('ExponentialMovingAverage')
+  return '/'.join(tf_name) + ':0'
+
+
+def check_match(keras_block, tf_block, keras_weights, tf_weights):
+  """ Check if the weights in h5 and ckpt match
+  
+  we match each name from keras_weights that is in keras_block 
+  and check if there is 1-1 correspondence to names from tf_weights
+  that is in tf_block
+  
+  Args:
+    keras_block: str, the block name for keras implementation (e.g. 'block1a')
+    tf_block: str, the block name for tf implementation (e.g. 'blocks_0')
+    keras_weights: list of str, each string is a name for weights in keras implementation
+    tf_weights: list of str, each string is a name for weights in tf implementation
+  """
+  for x in keras_weights:
+    if keras_block in x:
+      y = keras_name_to_tf_name_block(x, keras_block=bk, tf_block=bt)
+    match_lst.append(y)
+
+  assert len(match_lst) > 0
```

Comment:
Please provide a proper error message for the assert, otherwise will be very unreadable when this error happens.